src/App.tsx

```diff
@@ -15,48 +15,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type SortOptions = {
+  sortType: SortType,
+  reverse: boolean,
+};
+
+export function getPreparedGoods(
+  goods: string[],
+  { sortType, reverse }: SortOptions,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return SortType.NONE;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortField] = useState(SortType.NONE);
+  const [reverse, setSortReverse] = useState(false);
```

Comment:
Rename all boolean variables using [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isReverse, setIsReverse] = useState(false);
```